posthog/api/test/test_insight.py

```diff
@@ -1752,6 +1752,18 @@ def test_cancel_running_query(self) -> None:
         response = self.client.post(f"/api/projects/{self.team.id}/insights/cancel", {"client_query_id": f"testid"})
         self.assertEqual(response.status_code, 201, response.content)
 
+    @patch("posthog.decorators.get_safe_cache")
+    def test_including_query_id_does_not_affect_cache_key(self, patched_get_safe_cache) -> None:
```

Comment:
Is this in the right place?

We're basically testing filters hashing logic, so this test should logically be coupled along with other filter/query hashing tests and not depend on insights.

Reply:
> this test should logically be coupled along with other filter/query hashing tests and not depend on insights.

Unfortunately there's not really _a_ place where we test hashing/caching separately from insights. The two concepts are still very coupled. 

So, a general large font size +1 to making this tricky aspect of the system way more explicit but not sure this is the PR to do it

I'd rather use this PR only to see if removing this has an impact on cache hit ratio and how much.

------

Added a more specific test. Although (absent fixing how we generate the hash key) that test should be expanded to cover all the possible inputs and specify all the possible outputs